social-middleware/src/common/decorators/dev-only-doc.decorator.ts

```diff
@@ -0,0 +1,23 @@
+import { applyDecorators } from '@nestjs/common';
+import {
+  ApiOperation,
+  ApiQuery,
+  ApiResponse,
+  ApiExcludeEndpoint,
+} from '@nestjs/swagger';
+
+export function DevOnlySwaggerDocs() {
+  const isEnabled = process.env.ENABLE_DEV_TOOLS === 'true';
```

Comment:
should use configService.get<string> as per standard. (using NODE_ENV as per above)